test/Tester.cpp

```diff
@@ -0,0 +1,109 @@
+#include "../Json.h"
+#include "../Unit.h"
+#include "../Player.h"
+
+#include <gtest/gtest.h>
+
+TEST(UnitTest, ParseUnit)
+{
+    ASSERT_THROW(Unit::parseUnit("doesntexits.json"), std::runtime_error);
+    ASSERT_NO_THROW(Unit::parseUnit("../unit/unit1.json"));
+    Unit expected("Kakarott", 200, 90, 1.0);
+    Unit result = Unit::parseUnit("../unit/unit1.json");
+    EXPECT_EQ(expected.GetName(), result.GetName());
+    EXPECT_EQ(expected.GetDamage(), result.GetDamage());
+    EXPECT_EQ(expected.GetHealth(), result.GetHealth());
+    EXPECT_EQ(expected.GetCurrentCooldown(), result.GetCurrentCooldown());
+}
+TEST(UnitTest, ParseUnitMissingValue)
+{
+    ASSERT_THROW(Unit::parseUnit("./missingUnit.json"), std::runtime_error);
+}
+TEST(UnitTest, ParseUnitMinusValue)
+{
+    ASSERT_THROW(Unit::parseUnit("./minusValue.json"), std::runtime_error);
+}
+
+TEST(PlayerTest, Player)
```

Comment:
Szerintem a Player-rel kapcsolatots tesztek kerülhetnének külön fájlba, a jobb átláthatóság végett.